Task18 json

andresmadeddie opened the pull request:
Create read and write JSON methods using Jackson.
 JSON annotations on all classes in the model folder.
Use methods with 6 objects of different classes in the main method

andresmadeddie committed:
Create JsonUtils class with read and write Json files to a new folder called JSONFOlder
jsonWrite needs an object input, jsonRead needs the name of the file to read
Implement 5 object in the main method to use the previous methods

andresmadeddie committed:
Create JsonUtils class with read and write Json files to a new folder called JSONFOlder
jsonWrite needs an object input, jsonRead needs the name of the file to read
Implement 5 object in the main method to use the previous methods

andresmadeddie committed:
implement orders (6 in total) classes for JSON methods

andresmadeddie committed:
Json annotations to all classes on the model folder

andresmadeddie committed:
Json annotations to all classes on the model folder
 change to the correct date method

andresmadeddie committed:
minor changes
create a enums folder and move paths there

andresmadeddie committed:
minor changes
create a enums folder and move paths there

ankochergin commented on src/main/java/com/solvd/util/JSONExtractor.java:
```diff
@@ -0,0 +1,18 @@
+
+    public static String extractClassName(String fileName) {
+
+        Pattern pattern = Pattern.compile("^([a-z]|[A-Z])+");
```
```suggestion
    public static String extractClassName(String fileName) {
        Pattern pattern = Pattern.compile("^([a-z]|[A-Z])+");
```

ankochergin commented on src/main/java/com/solvd/util/JSONExtractor.java:
```diff
@@ -0,0 +1,18 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class JSONExtractor {
```
not sure that this is correct name for this utility class. I suppose you can get class name by filename and it can be not only json file. what do you think?

andresmadeddie replied:
I agree, I will rename it as ExtractFromString

ankochergin commented on src/main/java/com/solvd/sql/JJ/JSONUtils.java:
```diff
@@ -0,0 +1,46 @@
+            throw new RuntimeException();
+        }
+        return shop;
+    }*/
```
remove if not needed

andresmadeddie committed:
remove extra line between class and first field

Change class name from JSONExtractor to ExtractFromString

Remove unused code in JSONUtils

ankochergin commented on src/main/java/com/solvd/sql/JJ/JSONUtils.java:
```diff
@@ -0,0 +1,33 @@
+package com.solvd.sql.JJ;
```
rename package to `json` or `jackson`, remember about naming conventions

andresmadeddie committed:
Rename package JJ to jackson
Change name ExtractFromString to StringExtractor

andresmadeddie committed:
Rename package JJ to jackson
Change name ExtractFromString to StringExtractor

asurelaba commented on src/main/java/com/solvd/enums/Paths.java:
```diff
@@ -1,9 +1,10 @@
-    MODELFOLDER("com.solvd.sql.model");
+    MODELFOLDER("com.solvd.sql.model"),
+    JSONFOLDER("src/main/resources/JSONFolder/");
 
```
Liked the idea
